pkg/controllers/horizontalautoscaler/v1alpha1/suite_test.go

```diff
@@ -105,7 +105,7 @@ var _ = Describe("Examples", func() {
 		It("should scale to average value target, metric=41, target=4, want=11", func() {
 			Expect(ns.ParseResources("docs/examples/queue-length-average-value.yaml", ha, sng)).To(Succeed())
 			sng.Spec.Replicas = ptr.Int32(1)
-			fakeCloudProvider.NodeReplicas[sng.Spec.ID] = *sng.Spec.Replicas
+			fakeCloudProvider.NodeReplicas[sng.Spec.ID] = ptr.Int32(*sng.Spec.Replicas)
```

Comment:
mind adding a comment here about the race condition? Just to warn someone who might try to simplify it
```
fakeCloudProvider.NodeReplicas[sng.Spec.ID] = ptr.Int32(*sng.Spec.Replicas) # Create a new pointer to avoid races with the controller
```